tests/Silex/Tests/Application/ValidatorTraitTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Silex framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Silex\Tests\Application;
+
+use Silex\Application;
+use Silex\Provider\ValidatorServiceProvider;
+use Symfony\Component\Validator\Mapping\ClassMetadata;
+use Symfony\Component\Validator\Tests\Fixtures\Entity;
+use Symfony\Component\Validator\Tests\Fixtures\FakeMetadataFactory;
+use Symfony\Component\Validator\Tests\Fixtures\FakeClassMetadataFactory; // Symfony 2.1 only
+use Symfony\Component\Validator\Tests\Fixtures\FailingConstraint;
+
+/**
+ * ValidatorTrait test cases.
+ *
+ * @author Ludovic Fleury <ludo.fleur@gmail.com>
+ *
+ * @requires PHP 5.4
+ */
+class ValidatorTraitTest extends \PHPUnit_Framework_TestCase
+{
+    public function testValidate()
+    {
+        $app = $this->createApplication();
+
+        $stub = $this->getMock('TestClass');
+        $stub::staticExpects($this->any())
```

Comment:
``staticExpects`` was an experimental PHPunit feature which has been removed because it was not working properly